jax/lax/lax.py

```diff
@@ -1696,11 +1701,31 @@ def _sign_translation_rule(c, x):
 cos_p = standard_unop(_float | _complex, 'cos')
 ad.defjvp(cos_p, lambda g, x: neg(mul(g, sin(x))))
 
-atan2_p = standard_binop([_float, _float], 'atan2')
+atan2_p = standard_naryop([_float, _float], 'atan2')
 ad.defjvp(atan2_p,
   lambda g, x, y: _brcast(g, y) * (y / (square(x) + square(y))),
   lambda g, x, y: _brcast(g, x) * -x / (square(x) + square(y)))
 
+regularized_incomplete_beta_p = standard_naryop(
+    [_float, _float, _float], 'regularized_incomplete_beta')
+
+def betainc_gradx(g, a, b, x):
+  lbeta = lgamma(a) + lgamma(b) - lgamma(a + b)
+  partial_x = exp((b - 1) * log1p(-x) +
+                  (a - 1) * log(x) - lbeta)
+  g = _brcast(_brcast(g, a), b)
```

Comment:
Is this necessary? The primal operation didn't support broadcasting so why would the gradient need to?

Reply:
Note even when `lax` op did support broadcasting, it was done in the wrapper function, not in the primitive's implementation. So the primitive itself didn't broadcast and nor will its gradient need to.